src/cargo/ops/cargo_new.rs

```diff
@@ -925,3 +933,103 @@ fn update_manifest_with_inherited_workspace_package_keys(
         try_remove_and_inherit_package_key(key, manifest);
     }
 }
+
+/// Adds the new package member to the [workspace.members] array.
+/// - It first checks if the name matches any element in [workspace.exclude],
+///  and it ignores the name if there is a match.
+/// - Then it check if the name matches any element already in [workspace.members],
+/// and it ignores the name if there is a match.
+/// - If [workspace.members] doesn't exist in the manifest, it will add a new section
+/// with the new package in it.
+fn update_manifest_with_new_member(
+    root_manifest_path: &Path,
+    workspace_document: &mut toml_edit::Document,
+    package_path: &Path,
+) -> CargoResult<()> {
+    // Find the relative path for the package from the workspace root directory.
+    let workspace_root = root_manifest_path.parent().with_context(|| {
+        format!(
+            "workspace root manifest doesn't have a parent directory `{}`",
+            root_manifest_path.display()
+        )
+    })?;
+    let relpath = pathdiff::diff_paths(package_path, workspace_root).with_context(|| {
+        format!(
+            "path comparison requires two absolute paths; package_path: `{}`, workspace_path: `{}`",
+            package_path.display(),
+            workspace_root.display()
+        )
+    })?;
+
+    // Don't add the new package to the workspace's members
+    // if there is an exclusion match for it.
+    if let Some(members) = workspace_document
+        .get("workspace")
+        .and_then(|workspace| workspace.get("exclude"))
+        .and_then(|members| members.as_array())
+    {
+        for member in members {
+            let pat = member
+                .as_str()
+                .with_context(|| format!("invalid members' element `{}`", member))?;
+            if Path::new(pat) == relpath {
+                return Ok(());
+            }
+        }
+    }
+
+    let display_path = relpath
+        .to_str()
+        .map(|s| s.replace('\\', "/"))
+        .with_context(|| format!("invalid non-unicode path `{}`", relpath.display()))?;
```

Comment:
You likely want to do
```rust
let display_path: Result<String, _> = relpath.components().map(|os| os.to_str());
```
The key part being to use `components()` to split the path.  `\\` is a valid character in a path on Linux.